collectives/models/user.py

```diff
@@ -531,3 +532,17 @@ def activity_supervisors(activities):
     query = query.filter(Role.role_id == RoleIds.ActivitySupervisor)
     query = query.filter(User.id == Role.user_id)
     return query.all()
+
+
+def registered_users(event):
```

Comment:
Tu as déjà une fonction similaire dans event: `event.active_registrations()` qui pour chaque `registration` t'indiquera le `user`

Reply:
active_registrations me retourne une liste d'objets Registration.
Je n'ai que le user_id la dedans et pas d'email ?

Reply:
`event.active_registrations` te retourne une liste d'objet `collectives.models.registration.Registration`

Pour chaque registration, ça va te charger un `collectives.models.user.User` et tu pourras en donc accéder à l'email.

Pas trop lisible dans le code, ni dans la doc, mais tout le modèle et relations sont chargés :)

Donc la simple ligne suivante dans `send_cancelled_event_notification` suffit :)
 `       emails = [r.user.mail for r in event.active_registrations()]`